src/agent/mod.rs

```diff
@@ -434,6 +444,16 @@ where
         let client = self.client.as_ref().ok_or(OAuth2Error::NotInitialized)?;
         let config = self.config.as_ref().ok_or(OAuth2Error::NotInitialized)?;
 
+        //If specified in the login options will store the current url in the session for retrieval
+        // when coming back from IDP
+        if let Some(options) = config.options.clone() {
```

Comment:
I wonder if we actually need this to be configurable. It might be enough to just set this up in case the `redirect_url` differs from `current_url`

Reply:
That's an interesting idea. There might be a drawback (which could be handled as well of course). What would happen if one time that URL differs and the next time it doesn't? In that case, the URL would still be set and evaluated when coming back.

The question is: does it hurt to always store the URL when starting, and then to handle this when it comes back? If (after the code exchange) it detects a different URL than the current, it could do the post-login redirect. Unless a flag is present which could be used to prevent that (if people don't like it).

Reply:
My approach and idea when I started working on it was to set it only if it differs from redirect URL and then redirect to it if it's present when evaluating callback.

Reply:
So if I summarize:  
- The user set a fixed redirectUrl (LoginOptions)
- If the user has set a fixed redirectUrl then we store the currentUrl as postLoginUrl in the session
- Once the process is finished, if there is a postProcessUrl in the session , we redirect to it instead of the returnURL

Sounds like a good scenario.

Reply:
I'd reframe the second point'ts condition as 
- If the redirectUrl differs from the currentUrl
as that both means that redirect is needed and makes it obvious that fixed redirect url is set up

Reply:
Isn't this handled but the check at the third point? "If there is a postProcessUrl, we redirect to it instead of the returnUrl"?